sample-code/src/orchestration.ts

```diff
@@ -1,133 +1,170 @@
- * @returns The message content from the orchestration service in the generative AI hub.
+ * @returns The orchestration service response.
  */
-async function orchestrationCompletionTemplate(): Promise<string | undefined> {
+export async function orchestrationTemplating(): Promise<OrchestrationResponse> {
   const orchestrationClient = new OrchestrationClient({
-    llm: {
-      model_name: 'gpt-4-32k',
-      model_params: {}
-    },
+    llm: llmConfig,
     templating: {
       template: [
+        // define "country" as variable by wrapping it with "{{? ... }}"
         { role: 'user', content: 'What is the capital of {{?country}}?' }
       ]
     }
   });
 
-  // Call the orchestration service.
-  const response = await orchestrationClient.chatCompletion({
+  return orchestrationClient.chatCompletion({
+    // give the actual value for the variable "country"
     inputParams: { country: 'France' }
   });
-  // Access the response content.
-  return response.getContent();
 }
 
+const template = { template: [{ role: 'user', content: '{{?input}}' }] };
+
 /**
- * Allow any user input and apply input and output filters.
- * Handles the case where the input or output are filtered:
- * - In case the input was filtered the response has a non 200 status code.
- * - In case the output was filtered `response.getContent()` throws an error.
- * @returns The message content from the orchestration service in the generative AI hub.
+ * Apply a content filter to LLM requests, filtering any hateful input.
  */
-async function orchestrationCompletionFiltering(): Promise<string | undefined> {
+export async function orchestrationInputFiltering(): Promise<void> {
+  // create a filter with minimal thresholds for hate and violence
+  // lower numbers mean more strict filtering
   const filter = buildAzureContentFilter({ Hate: 0, Violence: 0 });
   const orchestrationClient = new OrchestrationClient({
-    llm: {
-      model_name: 'gpt-4-32k',
-      model_params: {}
-    },
-    templating: {
-      template: [{ role: 'user', content: '{{?input}}' }]
-    },
+    llm: llmConfig,
+    templating: template,
+    // configure the filter to be applied for both input and output
     filtering: {
-      input: filter,
-      output: filter
+      input: filter
     }
   });
 
   try {
-    // Call the orchestration service.
-    const response = await orchestrationClient.chatCompletion({
+    // trigger the input filter, producing a 400 - Bad Request response
+    await orchestrationClient.chatCompletion({
       inputParams: { input: 'I hate you!' }
     });
-    // Access the response content.
-    return response.getContent();
+    throw new Error('Input was not filtered as expected.');
   } catch (error: any) {
-    // Handle the case where the output was filtered.
-    return `Error: ${error.message}`;
+    if (error.response.status === 400) {
+      logger.info('Input was filtered as expected.');
+    } else {
+      throw error;
+    }
```

Comment:
Much better, I didn't like it the way I implemented this originally.
However:
[pp] Consider returning the logged info as a string here. Then in all other samples we could return the content from the response. Then the `getContent()` method would be more part of the sample, because it is in the same file as the rest of the code. Also we wouldn't need any special handling in the `server.ts` for the filtering cases.